num_enum_derive/src/lib.rs

```diff
@@ -241,37 +283,59 @@ impl Parse for EnumInfo {
                 Data::Struct(data) => die!(data.struct_token => "Expected enum but found struct"),
             };
 
-            let repr: Ident = {
-                let mut attrs = input.attrs.into_iter();
-                loop {
-                    if let Some(attr) = attrs.next() {
-                        if let Ok(Meta::List(meta_list)) = attr.parse_meta() {
-                            if let Some(ident) = meta_list.path.get_ident() {
-                                if ident == "repr" {
-                                    let mut nested = meta_list.nested.iter();
-                                    if nested.len() != 1 {
-                                        die!(attr =>
-                                            "Expected exactly one `repr` argument"
-                                        );
-                                    }
-                                    let repr = nested.next().unwrap();
-                                    let repr: Ident = parse_quote! {
-                                        #repr
-                                    };
-                                    if repr == "C" {
-                                        die!(repr =>
-                                            "repr(C) doesn't have a well defined size"
-                                        );
-                                    } else {
-                                        break repr;
-                                    }
-                                }
+            let attrs = input.attrs.into_iter();
+            let mut found_repr = None;
+            let mut crate_path = None;
+            for attr in attrs {
+                if attr.path.is_ident("num_enum") {
+                    match attr.parse_args_with(NumEnumAttributes::parse) {
+                        Ok(NumEnumAttributes { items }) => {
+                            if items.len() != 1 || crate_path.is_some() {
+                                die!(attr.tokens => "Expected exactly one `crate` argument");
+                            } else {
+                                let NumEnumAttributeItem::Crate(item) =
+                                    items.into_iter().next().unwrap();
+                                crate_path = Some(item.path);
+                            }
+                        }
+                        Err(err) => {
+                            die!(attr =>
+                                format!("Invalid attribute: {}", err)
+                            );
+                        }
+                    }
```

Comment:
This won't "scale well" if other attributes are added. I'd rather suggest an approach along the lines of `.flat_map()`ing / nested-`for`-style the attributes (since `#[num_enum(foo)] #[num_enum(bar)]` is the same as `#[num_enum(foo, bar)]`:

```rs
for attr in attrs {
    if attr.path.is_ident("num_enum") {
        for item in attr.parse_args_with(NumEnumAttributes::parse)? {
            match item {
                // when written like this, it's easier to extend with more attrs.
                NumEnumAttributeItem::Crate(item) => {
                    if crate_path.replace(item.path).is_some() {
                        die!(item.krate => "duplicate `crate` attribute");
                    }
                },
            }
        }
    }
    …
}
```